### PR TITLE
SAS-69# unit test for user and sub classes

### PR DESCRIPTION
**Pull request by yaarm**
Written by Chen and I.
Unit test for user and all sub classes.

### CODE REVIEW

File: src/main/java/com/SAS/User/Coach.java [hunk 1]
```diff
@@ -42,7 +42,7 @@ public User getUser() {
      *
      * @return level - int
      */
-    public Integer getLevel() {
+    public int getLevel() {
         return level;
     }
```
Comment (MrBahur) [inline src/main/java/com/SAS/User/Coach.java:48]: why did you change Integer to int?

File: src/main/java/com/SAS/User/Coach.java [hunk 2]
```diff
@@ -61,7 +61,9 @@ public FieldRole getFieldRole() {
      * @param level - int
      */
     public void setLevel(int level) {
-        this.level = level;
+        if (level > 0) {
+            this.level = level;
+        }
     }
```
Comment (MrBahur) [inline src/main/java/com/SAS/User/Coach.java:71]: what about feedback when level<=0?

File: src/main/java/com/SAS/User/Coach.java (hunk 1 above)
Comment (yaarm) [reply src/main/java/com/SAS/User/Coach.java:48]: Because it shouldn't be an Integer, there is not reason.

File: src/main/java/com/SAS/User/Coach.java (hunk 2 above)
Comment (yaarm) [reply src/main/java/com/SAS/User/Coach.java:71]: I'll fix it.

File: src/main/java/com/SAS/User/Coach.java [hunk 3]
```diff
@@ -102,11 +108,15 @@ public void removeAssetFromTeam() {
...
      * The function adds personal page to the coach
-     *
      * @param description
+     * @return pageID - int
      */
-    public void addPage(String description) {
-        this.personalPage = new PersonalPage(description);
+    public int addPage(String description) {
+        if (description != null && !description.trim().isEmpty()) {
+            this.personalPage = new PersonalPage(description);
```
Comment (MrBahur) [inline src/main/java/com/SAS/User/Coach.java:119]: line 115, those are 2 different erros, you should split it to 2 different cases in my opinion

File: src/main/java/com/SAS/User/Coach.java [hunk 4]
```diff
@@ -140,18 +153,22 @@ public String getRole() {
...
+
         try{
             level = Integer.parseInt(details.get(0));
+            fieldRole = convertStringToFieldRole((details.get(1)));
         }catch (Exception e){
 
         }
+
         if (fieldRole == null || level == -1) {
             return false;
```
Comment (MrBahur) [inline src/main/java/com/SAS/User/Coach.java:170]: same here as line 115

File: src/main/java/com/SAS/User/UserController.java [hunk 5]
```diff
@@ -1,13 +1,14 @@
...
-import java.util.HashSet;
+import com.SAS.crudoperations.CRUD;
 
 public class UserController {
 
     /**
      * singleton Privileges instance
      */
     private Privileges globalPrivileges;
+    private CRUD crud;
```
Comment (MrBahur) [inline src/main/java/com/SAS/User/UserController.java]: update this because now Crud is static and you don't need instance of it to use it's functions

File: src/main/java/com/SAS/User/UserController.java [hunk 6]
```diff
@@ -16,14 +17,34 @@ public UserController() {
...
+    public User createUser(String userName, String password, String fullName, UserType type, boolean approval, User newUser){
+         //check user parameters
+         if (!validateDetails(userName, password, fullName, type)) {
+             System.out.println("The parameters aren't valid. Pleas enter the details again.");
+             return newUser;
+         }
+
+         //checks if the userName already exists
+         if (crud.isUserNameExist(userName)) {
+             System.out.println("The userName already exist. Please enter your details again.");
```
Comment (MrBahur) [inline src/main/java/com/SAS/User/UserController.java:42]: update this because now CRUD is static and you don't need instance of it to use it's functions

File: src/main/java/com/SAS/User/Coach.java (hunk 3 above)
Comment (yaarm) [reply src/main/java/com/SAS/User/Coach.java:119]: Why? it's the same case for me.

**Review by MrBahur — CHANGES_REQUESTED** [review]
also, please add the updated CRUD and pom.xml and then merge.

File: src/main/java/com/SAS/User/Coach.java (hunk 1 above)
Comment (MrBahur) [reply src/main/java/com/SAS/User/Coach.java:48]: the Integer class is provided so that values can be boxed/unboxed in a pure OO manner. use int where appropriate unless you specifically need to use it in an OO way; in which case Integer is appropriate

However, very different things are going on under the covers here. An int is a number
 an > Integer is a pointer that can reference an object that contains a number

An int is not an object and cannot passed to any method that requires objects. A common case is in using the provided collection classes ( List , Map , Set ) - though it is possible to write versions of these classes that provide similar capabilities to the object versions. The wrapper classes ( Integer , Double , etc) are frequently required whenever introspection is used (such as in the reflection API)

concider that and update me on your decission :)

File: src/main/java/com/SAS/User/Coach.java (hunk 3 above)
Comment (MrBahur) [reply src/main/java/com/SAS/User/Coach.java:119]: because if the description is null, it didn't initiated.
if the description is empty string, it initiated but the vulue didn't saved.

File: src/main/java/com/SAS/User/UserController.java (hunk 5 above)
Comment (yaarm) [reply src/main/java/com/SAS/User/UserController.java]: I'll fix it.

File: src/main/java/com/SAS/User/Coach.java (hunk 3 above)
Comment (yaarm) [reply src/main/java/com/SAS/User/Coach.java:119]: I understand, but in the two cases I don't want to add it to the personal page.

File: src/main/java/com/SAS/User/UserController.java (hunk 6 above)
Comment (yaarm) [reply src/main/java/com/SAS/User/UserController.java:42]: I'll fix it.

File: src/main/java/com/SAS/User/Coach.java (hunk 1 above)
Comment (yaarm) [reply src/main/java/com/SAS/User/Coach.java:48]: I think I'll leave like this, since it not a collection.

File: src/main/java/com/SAS/User/Coach.java (hunk 3 above)
Comment (Chen2908) [reply src/main/java/com/SAS/User/Coach.java:119]: In our opinion, both cases are invalid input so we wouldn't want to create a personal page which is blank. 
Therefore we would leave the implementation this way :)

Comment (MrBahur) [reply src/main/java/com/SAS/User/Coach.java:119]: but the handale should be different, if there is a problem with null its different problem from empty string, you are wrong.

File: src/main/java/com/SAS/User/Coach.java (hunk 1 above)
Comment (MrBahur) [reply src/main/java/com/SAS/User/Coach.java:48]: i only gave you one example, but as you wish.

**Review by MrBahur — APPROVED** [review]
